optimum/neuron/modeling_decoder.py

```diff
@@ -0,0 +1,208 @@
+
+        if task is None:
+            task = TasksManager.infer_task_from_model(cls.auto_model_class)
+
+        # Instantiate the exporter for the specified configuration and task
+        exporter = get_exporter(config, task)
+
+        # Split kwargs between model and neuron args
+        model_kwargs, neuron_kwargs = exporter.split_kwargs(**kwargs)
+
+        # Instantiate the transformers model checkpoint
+        model = TasksManager.get_model_from_task(
+            task=task,
+            model_name_or_path=model_id,
+            subfolder=subfolder,
+            revision=revision,
+            framework="pt",
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            **model_kwargs,
+        )
+
+        # Save the model checkpoint in a temporary directory
+        checkpoint_dir = TemporaryDirectory()
+        save_pretrained_split(model, checkpoint_dir.name)
+
+        # Update the config
+        config.neuron = {"task": task, "neuron_kwargs": neuron_kwargs}
+
+        return cls._from_pretrained(checkpoint_dir, config)
+
+    @classmethod
+    def _get_neuron_paths(cls, model_dir: Union[str, Path, TemporaryDirectory]) -> Tuple[str, str, str]:
+        if isinstance(model_dir, TemporaryDirectory):
+            model_path = model_dir.name
+            # We are in the middle of an export: the checkpoint is in the temporary model directory
+            checkpoint_path = model_path
+            # There are no compiled artifacts yet
+            compiled_path = None
+        else:
+            model_path = model_dir
+            # The model has already been exported, the checkpoint is in a subdirectory
+            checkpoint_path = os.path.join(model_path, cls.CHECKPOINT_DIR)
+            # So are the compiled artifacts
+            compiled_path = os.path.join(model_path, cls.COMPILED_DIR)
+        return model_path, checkpoint_path, compiled_path
+
+    @classmethod
+    def _from_pretrained(
+        cls, model_id: Union[str, Path, TemporaryDirectory], config: "PretrainedConfig", **kwargs
+    ) -> "NeuronDecoderModel":
+        # Verify we are actually trying to load a neuron model
+        neuron_config = getattr(config, "neuron", None)
+        if neuron_config is None:
+            raise ValueError(
+                "The specified directory does not contain a neuron model. "
+                "Please convert your model to neuron format by passing export=True."
+            )
+
+        # Evaluate the configuration passed during export
+        task = neuron_config["task"]
+        neuron_kwargs = neuron_config["neuron_kwargs"]
+
+        exporter = get_exporter(config, task)
+
+        model_path, checkpoint_path, compiled_path = cls._get_neuron_paths(model_id)
+
+        neuronx_model = exporter.neuronx_class.from_pretrained(checkpoint_path, **neuron_kwargs)
```

Comment:
For optimum modeling, we usually put the compilation under `from_transformers`, and `from_pretrained` will either load a pre-compiled model or compile it by calling `from_transformers`.

Reply:
This is what i did at first, but then I realized I need to apply the "modeling/transform" step also when loading a model from a local path. This is why it happens only later.